Alignment/OfflineValidation/scripts/validateAlignments.py

```diff
@@ -3,769 +3,393 @@
-        self.validation.createConfiguration( outpath )
-        if self.preexisting:
-            return
-        self.scripts = sum([addIndex(script, self.validation.NJobs) for script in self.validation.createScript( outpath )], [])
-        return None
-
-    def createJob(self):
-        """This is the method called to create the job files."""
-        self.__createJob( self.validation.jobmode,
-                          os.path.abspath( self.commandLineOptions.Name) )
-
-    def runJob( self ):
 
-        general = self.config.getGeneral()
-        log = ""
+import Alignment.OfflineValidation.TkAlAllInOneTool.GCP as GCP
+import Alignment.OfflineValidation.TkAlAllInOneTool.DMR as DMR
+import Alignment.OfflineValidation.TkAlAllInOneTool.PV as PV
+import Alignment.OfflineValidation.TkAlAllInOneTool.SplitV as SplitV
+import Alignment.OfflineValidation.TkAlAllInOneTool.JetHT as JetHT
+
+##############################################
+def parser():
+##############################################
+    """ Parse user input """
+
+    parser = argparse.ArgumentParser(description = "AllInOneTool for validation of the tracker alignment", formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("config", metavar='config', type=str, action="store", help="Global AllInOneTool config (json/yaml format)")
+    parser.add_argument("-d", "--dry", action = "store_true", help ="Set up everything, but don't run anything")
+    parser.add_argument("-v", "--verbose", action = "store_true", help ="Enable standard output stream")
+    parser.add_argument("-e", "--example", action = "store_true", help ="Print example of config in JSON format")
+    parser.add_argument("-f", "--force", action = "store_true", help ="Force creation of enviroment, possible overwritten old configuration")
+    parser.add_argument("-j", "--job-flavour", action = "store", default = "longlunch", choices = ["espresso", "microcentury", "longlunch", "workday", "tomorrow", "testmatch", "nextweek"], help ="Job flavours for HTCondor at CERN, default is 'longlunch'")
+
+    return parser.parse_args()
+
+##############################################
+def digest_path(path):
+##############################################
+    """ Expand environment variables in the input string
+    Arguments:
+    - path: String from which environment variables are expanded
+    Return:
+    - the input string with environment variables expanded
+    """
+    # split path in folders
+    path_s = str(path).split(os.sep)
+
+    path_d_s = []
+    for part in path_s:
+        # Look for environment variables such as $CMSSW_BASE
+        if part.startswith('$'):
+            env_var = part[1:].replace('{', '').replace('}', '')
+            path_d_s.append(os.environ[env_var])
+        else: path_d_s.append(part)
+
+    # re join folders in to a path
+    path_d = os.path.join(*path_d_s)
+
+    # re add front / if needed
+    if path.startswith(os.sep): path_d = os.sep + path_d
+
+    return path_d
```

Comment:
This kind of function (or similar) needs to be re-defined in each of the `Alignment/OfflineValidation/python/TkAlAllInOneTool/<Validation_Name>.py` in order to properly account for any environmental variable that might be used. This is not done automatically so this function must be placed by hand wherever user might want to use env. variable. So in case we remove lumi files from local data directory we need to extend this function to loop over possible locations. That's what I wanted to say.

Reply:
@smuzaffar is there a better way to deal with the equivalent of `edm::FileInPath` in a python script?

Reply:
yes e.g. see https://github.com/cms-sw/cmssw/blob/master/FWCore/ParameterSet/python/pfnInPath.py

Reply:
thanks @smuzaffar!
@TomasKello please try to adapt the script to use `pfnInPath` and don't forget to remove the ASCII files from the data directory and put them in the cms-data one.

Reply:
Well, that's what we are already doing, for instance [here](https://github.com/cms-sw/cmssw/pull/38304/files#diff-6f60d4c67f2ef913f5719906bb41137c9f360c4d939853cc82c0ad4791ae647dR537). That function above is the underlying problem I try to explain. Local configuration will not accept environmental variables unless they are explicitly translated into string in DMR.py, PV.py, JetHT.py, GCP.py, etc. But as I have some more time now before release is closed I can still implement some global solution.

Reply:
> However, with lumifiles now removed from local /data dir and cms-data PR not yet merged, all tests will fail for this PR for sure ...

That's not an issue, we can test together with that